shuup/reporting/utils.py

```diff
@@ -0,0 +1,99 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+import datetime
+import decimal
+
+import babel
+# import dateparser
+import six
+from babel.numbers import format_currency
+from django.utils import timezone
+from django.utils.translation import get_language
+
+ERROR_MESSAGE = "Error fetching data"
+NO_DATA_MESSAGE = "No results"
+
+
+def get_empty_data(schema, data, message):
+    keys = [s.get("key") for s in schema]
+    return_data = {}
+    for key in keys:
+        return_data[key] = message
+    return {
+        "data": [return_data],
+        "start": data["start"],
+        "end": data["end"],
+    }
+
+
+def get_error_data(schema, sales_data):
+    if sales_data.get("data") == "error":
+        return get_empty_data(schema, sales_data, ERROR_MESSAGE)
+    if not sales_data.get("data")[0]:
+        return get_empty_data(schema, sales_data, NO_DATA_MESSAGE)
+
+
+def to_datetime(value):
+    # TODO: Use this?
+    # if isinstance(value, six.string_types):
+    #     return dateparser.parse(value)
+
+    if isinstance(value, datetime.datetime):
+        return value
+
+    if isinstance(value, datetime.date):
+        return datetime.datetime.combine(value, datetime.time())
+
+    return None
+
+
+def parse_date_range_preset(value):
+    now = timezone.now()
+    if value == "today":
+        midnight = now.replace(hour=0, minute=0, second=0)
+        tomorrow = midnight + datetime.timedelta(days=1)
+        return (midnight, tomorrow)
+    if value == "running_week":
+        return (now - datetime.timedelta(days=7), now)
+    if value == "running_month":
+        return (now - datetime.timedelta(days=30), now)
+    if value == "this_month":
+        return (now.replace(day=1), now)
+    if value == "this_year":
+        return (now.replace(day=1, month=1), now)
+    if value == "all_time":
+        return (now.replace(year=2000), now)
+
+
+def parse_date_range(value):
+    output = parse_date_range_preset(value)
+    if output:
+        return output
+
+    if isinstance(value, six.string_types):
+        value = value.strip()
+        if ".." in value:
+            start, end = value.split("..", 1)
+        elif value.count("-") == 1:
+            start, end = value.split("-", 1)
+        else:
+            start, end = value.split(None, 1)
+    elif isinstance(value, (list, tuple)):
+        start, end = value[:2]
+    else:
+        raise ValueError("Can't split date range: %r" % value)
+    date_range = (to_datetime(start), to_datetime(end))
+    if any(p is None for p in date_range):
+        raise ValueError("Invalid date range: %r (parsed as %r)" % (value, date_range))
+    return date_range
+
+
+def format_data_with_usd_currency(sales_data):
```

Comment:
Where is this used? Do we really need this. Should we use `shuup.utils.i18n` for these kind of things?